components/kyma-environment-broker/internal/broker/client.go

```diff
@@ -114,25 +118,61 @@ func (c *Client) SendExpirationRequest(instance internal.Instance) (string, erro
 	}
 	defer c.warnOnError(resp.Body.Close)
 
-	processStatusCode(resp.StatusCode)
+	return processResponse(resp.StatusCode, resp)
+}
 
-	operation, err := decodeResponseOperation(err, resp)
-	if err != nil {
-		return "", err
+func processResponse(statusCode int, resp *http.Response) (string, error) {
+	switch statusCode {
+	case http.StatusAccepted, http.StatusOK:
+		{
+			log.Infof("Request accepted with status: %+v", statusCode)
+			operation, err := decodeOperation(resp)
+			if err != nil {
+				return "", err
+			}
+			log.Infof("operation: %+v", operation)
+			return operation, nil
+		}
+	case http.StatusUnprocessableEntity:
+		{
+			log.Warnf("Entity unprocessable - request rejected with status: %+v", statusCode)
+			description, errorString, err := decodeErrorResponse(resp)
+			if err != nil {
+				return "", err
```

Comment:
perhaps wrap the error for more context?

Reply:
Error is wrapped in decodeErrorResponse function.